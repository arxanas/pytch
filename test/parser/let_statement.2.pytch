let foo =
    let bar =
        1
    bar

let baz = 3
