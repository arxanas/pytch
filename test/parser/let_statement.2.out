SyntaxTree
    LetExpr
        Token 'let'
        VariablePattern
            Leading ' '
            Token 'foo'
        Leading ' '
        Token '='
        Trailing '\n'
        LetExpr
            Leading '    '
            Token 'let'
            VariablePattern
                Leading ' '
                Token 'bar'
            Leading ' '
            Token '='
            Trailing '\n'
            IntLiteralExpr
                Leading '        '
                Token '1'
                Trailing '\n'
            Token DUMMY_IN ''
            IdentifierExpr
                Leading '    '
                Token 'bar'
                Trailing '\n'
                Trailing '\n'
        Token DUMMY_IN ''
        LetExpr
            Token 'let'
            VariablePattern
                Leading ' '
                Token 'baz'
            Leading ' '
            Token '='
            IntLiteralExpr
                Leading ' '
                Token '3'
                Trailing '\n'
            Token DUMMY_IN ''
            <missing>
    Token EOF ''
